src/dbnode/client/write_state.go

```diff
@@ -53,22 +65,25 @@ type writeState struct {
 	sync.Mutex
 	refCounter
 
-	consistencyLevel                     topology.ConsistencyLevel
-	shardsLeavingCountTowardsConsistency bool
-	topoMap                              topology.Map
-	op                                   writeOp
-	nsID                                 ident.ID
-	tsID                                 ident.ID
-	tagEncoder                           serialize.TagEncoder
-	annotation                           checked.Bytes
-	majority, pending                    int32
-	success                              int32
-	errors                               []error
-	lastResetTime                        time.Time
-
-	queues         []hostQueue
-	tagEncoderPool serialize.TagEncoderPool
-	pool           *writeStatePool
+	consistencyLevel                                       topology.ConsistencyLevel
+	shardsLeavingCountTowardsConsistency                   bool
+	shardsLeavingAndInitializingCountTowardsConsistency    bool
+	successAsLeavingAndInitializingCountTowardsConsistency bool
+	topoMap                                                topology.Map
+	reusableByteID                                         *ident.ReusableBytesID
```

Comment:
why do you need this? you already get host IDs as ident.ID from shard - no?

Reply:
in that case - maybe just use `ident.ID` here? (and construct the value using `ident.StringID(...)`)

Reply:
you can make hostSuccessList a `[]string` (because as you point out - you're already getting hosts back as `string` from the other codepaths). that'll avoid all this hassle